modern/src/SocketController.js

```diff
@@ -83,7 +74,32 @@ const SocketController = () => {
     return null;
   }, [authenticated]);
 
-  return null;
+  useEffect(() => {
+    setNotifications(events.map((event) => ({
+      id: event.id,
+      message: `${devices[event.deviceId]?.name}: ${t(prefixString('event', event.type))}`,
+      show: true,
+      close: () => setEvents(events.filter((e) => e.id !== event.id)),
```

Comment:
I don't like that there's a callback in a data object. Why do we need it here? We can just call it directly in `onClose`.

Reply:
Done.